tests/test_u_train.py

```diff
@@ -38,3 +41,206 @@ def test_u_train():
 
     br = linker._settings_obj._blocking_rules_to_generate_predictions[0]
     assert br.blocking_rule == "l.name = r.name"
+
+
+def test_u_train_link_only():
```

Comment:
This test should be more than sufficient to assess whether this works for all linkers fyi. 

There's no need to do separate checks for spark, athena and sqlite in my view.